ext/session/session.c

```diff
@@ -637,11 +652,43 @@ static PHP_INI_MH(OnUpdateName) /* {{{ */
 		return FAILURE;
 	}
 
-	OnUpdateStringUnempty(entry, new_value, mh_arg1, mh_arg2, mh_arg3, stage);
-	return SUCCESS;
+	return OnUpdateStringUnempty(entry, new_value, mh_arg1, mh_arg2, mh_arg3, stage);
+}
+/* }}} */
+
+
+static PHP_INI_MH(OnUpdateCookieLifetime) /* {{{ */
+{
+	SESSION_CHECK_OUTPUT_STATE;
+	return OnUpdateLongGEZero(entry, new_value, mh_arg1, mh_arg2, mh_arg3, stage);
```

Comment:
Negative lifetime is treated the same as 0. It is mitigation for integer overflow that user would like to set very long lifetime, but it overflowed by computation. It's unlikely, so it could be OnUpdateLong. Should I?